site/app/libraries/socket/Server.php

```diff
@@ -38,11 +38,18 @@ public function __construct(Core $core) {
     private function checkAuth(ConnectionInterface $conn): bool {
         $request = $conn->httpRequest;
         $client_id = $conn->resourceId;
-        $userAgent = $request->getHeader('user-agent');
+        $origin = $request->getHeader('origin')[0];
+        $user_agent = $request->getHeader('user-agent')[0];
 
-        if ($userAgent[0] === "websocket-client-php") {
+        if ($user_agent == 'python-websocket-client') {
+            // pass authentication when testing
             return true;
         }
+
+        if (!strpos($origin, "localhost")) {
```

Comment:
If I am not mistaken, we would want this check to return true right (and check if the request is coming from localhost)? If the request is coming from localhost is it not presumed that the service making the request is being ran locally? Please let me know if this is incorrect